### DISCUSSION TITLE
How to create page with multi-layouts for a cross-platform app ?

### DISCUSSION THREAD

**Discussion by valentin-harrang** [vercel/next.js#53559] [Help]
### Summary

Hello,

I'd like to use Next.js 13 to create a cross-platform application.

Each platform will have its own domain name and each page will have a different layout depending on the domain. So I'm thinking of creating a middleware that will redirect the domain to the `app/[domain]` folder but I don't think I have any choice but to make a switch case in each layout to know which layout to use. 

This example is very good, but there is no layout for each platform: https://vercel.com/guides/nextjs-multi-tenant-application

My questions:

- How do you link a layout to a domain?
- Is it possible to do this dynamically without a switch case?
- And if I need to make conditions, is it better to base them on the current domain by accessing the `params.domain` property or on an environment variable?

Folder structure:

<img width="355" alt="image" src="https://github.com/vercel/next.js/assets/19478137/46294ae3-6476-47e9-ae52-19c02692d398">

middleware.ts:

```js
import { NextRequest, NextResponse } from "next/server";

export const config = {
  matcher: [
    /*
     * Match all paths except for:
     * 1. /api routes
     * 2. /_next (Next.js internals)
     * 3. /_static (inside /public)
     * 4. all root files inside /public (e.g. /favicon.ico)
     */
    "/((?!api/|_next/|_static/|_vercel|[\\w-]+\\.\\w+).*)",
  ],
};

export default async function middleware(req: NextRequest) {
  const { nextUrl, headers, url } = req;

  const hostname = headers.get("host")!;
  const path = nextUrl.pathname;

  return NextResponse.rewrite(new URL(`/${hostname}${path}`, url));
}
```

If you have a better idea, don't hesitate to let me know.

Thank you

**Comment by Nadeera3784** [comment]
Hi @valentin-harrang How about using Dynamic Layout?

```
import { useEffect } from 'react';
import dynamic from 'next/dynamic';

// Dynamic import of the layout component
const Layout = dynamic(() => import(`./${process.env.NEXT_PUBLIC_DOMAIN}/layout`));

export default function PlatformPage() {
  useEffect(() => {
    // Your domain-specific code here
  }, []);

  return (
    <Layout>
      {/* Your domain-specific content here */}
    </Layout>
  );
}
```

**Reply by valentin-harrang** [reply]
Very nice ! But it's on client side ?
